### ISSUE TITLE
Cover status reporting

### ISSUE THREAD

**Issue by remygrandin** [Den901/ha_came#9]
Hi,

I installed the pluggin and everything works, apart from the cover status, it is always showing as "unknown".

I digged in the cover code and saw the entity exposed 'is_closing', 'is_opening' and 'is_stopped', but they are not in use by the pluggin, also, they seem to be showing always as false, false and true, respectivly, whatever the blind real state is.

I know in my installation the cover don't have a progress detector, but I see on the sapiens web interface that it know what "state" it is in (closing/openning/stopped) by the blinking button it show.

![image](https://github.com/user-attachments/assets/062d00ed-873a-458c-a883-4508dce090df)

Can this status be reported to HA ?

PS : Thanks for this plugin, I can finally use the domo provided with my flat :-)

**Comment by Den901** [comment]
Yes, we can work to add the status (honestly idk why he didn't memorize the last movement 🤣), i need a test cover for try if the code is working.

As u know, the percentage is not supported by came and adding it using the time would be create the caos...

I will investigate on it

**Comment by remygrandin** [comment]
Yeah, I hope one day to be able to either add an external zigbee course tracker, or maybe try to do a computed course estimate based on activation durations, but directly on HA side of course.

If you need some HAR requests file from the came web interface I can provide them. I stay available also for any test you wish :-)

**Comment by Den901** [comment]
Yep! Do you have the came api? I'm curious to request a cover status to the domo.....

**Comment by Den901** [comment]
Alternatively i can send u the api

**Comment by remygrandin** [comment]
I dont have the ref, and only the pyc files of the pycame lib, no idea how pip was able to access those for install with your doc link :-)
If you have the ref, I can query it

**Comment by Den901** [comment]
Write me an email maurodanny90@gmail.com, i can't post them here

**Comment by remygrandin** [comment]
I confirm the api respose with a status depending on the cover movement, respectively 0/1/2 for stopped, openning, closing
(json edited to only 1 cover)

stopped : 
```
{
"cseq": 1,
"cmd_name": "openings_list_resp",
"array": [{
"open_act_id": 34,
"close_act_id": 35,
"name": "Bureau",
"floor_ind": 17,
"room_ind": 853,
"status": 0,
"partial": [],
"type": 0
}],
"sl_data_ack_reason": 0
}
```

openning :
```
{
"cseq": 2,
"cmd_name": "openings_list_resp",
"array": [ {
"open_act_id": 34,
"close_act_id": 35,
"name": "Bureau",
"floor_ind": 17,
"room_ind": 853,
"status": 1,
"partial": [],
"type": 0
}],
"sl_data_ack_reason": 0
}
```

Closing:
```
{
"cseq": 2,
"cmd_name": "openings_list_resp",
"array": [ {
"open_act_id": 34,
"close_act_id": 35,
"name": "Bureau",
"floor_ind": 17,
"room_ind": 853,
"status": 2,
"partial": [],
"type": 0
}],
"sl_data_ack_reason": 0
}
```

**Comment by Den901** [comment]
Ok this week i will take a look 🫡

**Comment by remygrandin** [comment]
I've updated to 1.0.4 but the issue is till there, they still report as unknown

**Comment by Den901** [comment]
The cover status it's not available on CAME API, cause it's not supported, we can add the open/close status after sending a command but it's a fake and it could generate other issues...

**Comment by remygrandin** [comment]
What is then the "status " i got above ? It seem to match the stopped, openning, closing and the main cale web ui is able to show this

**Comment by Den901** [comment]
I have to investigate, cause homo sapiens and came domotic may have some differences between the API

Inviato da Outlook per iOS<https://aka.ms/o0ukef>
________________________________
Da: remygrandin ***@***.***>
Inviato: Thursday, July 3, 2025 8:53:30 AM
A: Den901/ha_came ***@***.***>
Cc: Danny Mauro ***@***.***>; State change ***@***.***>
Oggetto: Re: [Den901/ha_came] Cover status reporting (Issue #9)

[https://avatars.githubusercontent.com/u/1934515?s=20&v=4]remygrandin left a comment (Den901/ha_came#9)<https://github.com/Den901/ha_came/issues/9#issuecomment-3031089777>

What is then the "status " i got above ? It seem to match the stopped, openning, closing and the main came web ui is able to show this
